src/CommandBus/Handler/DefaultParameterResolver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\CommandBus\Handler;
+
+use Patchlevel\EventSourcing\Attribute\Inject;
+use Psr\Container\ContainerInterface;
+use ReflectionMethod;
+use ReflectionParameter;
+use Symfony\Component\TypeInfo\Type\ObjectType;
+use Symfony\Component\TypeInfo\TypeResolver\TypeResolver;
+
+final class DefaultParameterResolver implements ParameterResolver
+{
+    public function __construct(
+        private readonly ContainerInterface|null $container = null,
+    ) {
+    }
+
+    /** @return iterable<mixed> */
+    public function resolve(ReflectionMethod $method, object $command): iterable
+    {
+        foreach ($method->getParameters() as $index => $parameter) {
+            if ($index === 0) {
+                yield $command; // first parameter is always the command
+
+                continue;
+            }
+
+            if (!$this->container) {
+                throw ServiceNotResolvable::missingContainer();
+            }
+
+            yield $this->container->get(self::serviceName($method, $parameter));
```

Comment:
Maybe a try catch the exception from `container->get()` and re-throw the exception with `ServiceNotResolvable::missingService($method, $parameter, $exception)` ?